src/kakarot/instructions/system_operations.cairo

```diff
@@ -816,12 +816,14 @@ namespace CreateHelper {
         // so we use popped_len to derive the way we should handle
         // the creation of evm addresses
         if (popped_len != 4) {
-            let (tx_info) = get_tx_info();
+            let (nonce) = IContractAccount.get_nonce(ctx.starknet_contract_address);
```

Comment:
For `CREATE`, we take the nonce of the contract account

Reply:
I remember a discussion regarding whether we increment before of after actually computing the `evm_address`, posting this just to make sure that you checked

Reply:
I don't really see the point with collision, are you sure? @jobez we discovered this nonce update before or after created an issue with forge (precomputed address was not the right one), but I don't know if we actually have a test for this

Reply:
If I understand it correctly for `CREATE` it uses nonce, so if there are several txs with `CREATE` being sent, then we will have address collision because we deploy first before incrementing the nonce, won't we?

Reply:
Ah sorry I see what you mean, I've moved incrementing the once to after we compute the `evm_address` and before we actually create the account